src/serac/physics/materials/thermal_functional_material.hpp

```diff
@@ -88,6 +88,89 @@ class LinearIsotropicConductor {
   double conductivity_;
 };
 
+/// Linear isotropic thermal conduction material model
+class ParameterizedLinearIsotropicConductor {
+public:
+  /**
+   * @brief Construct a new Parameterized Linear Isotropic Conductor object
+   *
+   * @param density Density of the material (mass/volume)
+   * @param specific_heat_capacity Specific heat capacity of the material (energy / (mass * temp))
+   * @param conductivity Thermal conductivity of the material (power / (length * temp))
+   */
```

Comment:
This is more of an example than practical code, right?

Reply:
Could you clarify which aspects seem impractical to you? Jamie and I have been discussing a different way to express these materials that cuts out a lot of the boilerplate present in this implementation, but if you have other constructive criticisms, we'd like to hear them!

Reply:
Yes - this material isn't particularly useful. I'll had some materials where the conductivity/loads are directly specified by a parameter.

Reply:
> Could you clarify which aspects seem impractical to you? Jamie and I have been discussing a different way to express these materials that cuts out a lot of the boilerplate present in this implementation, but if you have other constructive criticisms, we'd like to hear them!

In a practical example, I would expect:

- semantic naming of the parameter to make it's meaning clear to the engineers who would use it
- presumably the units or that it's dimensionless should be given for the parameter
- the use of the parameter in the calculation of function values should make physical sense